Wrapping up documentation and some mild testing (#7)
getlantern/geneva-cli: README.md

`go build`

Additionally you can install the program by adding the folder to your powershell path. 
This will allow you to call with just geneva-cli instead of .\geneva-cli.exe from anywhere. 

Any filepaths you paths should be absolute paths, the working directory will be changed to the executable directory.

## How to run
First you will need a valid Geneva strategy, one is included in s.txt.

Then you can run the program using

`.\geneva-cli.exe intercept --interface <interface-name> -strategyFile .\s.txt`

or

`.\geneva-cli.exe intercept --interface <interface-name> -strategy "[TCP:flags:PA]-fragment{tcp:-1:True}-| \/"`

> [TCP:flags:PA]-fragment{tcp:8:True}(,fragment{tcp:4:True})-| \/

Reviewer: What is this line?

Author: A mistake, I had the wrong window open at some point trying to copy a strategy to console

`.\geneva-cli.exe list-adapters`

## How to test

> You can test or rather verify that the intercept mode is running by simply running it with a validated strategy. The output should show that packets are being rerouted. You can monitor your adapter in [WireShark](https://www.wireshark.org/) and you should notice a huge uptick in packets being sent from your adapter.

Reviewer: The only strategy I've been able to run is the one in s.txt.  Which other strategies did you test with?  Also many strategies generate error statements, but the reference is malformed and the error is `nil`.  Here's an example using `[TCP:flags:S]-duplicate(tamper{TCP:flags:replace:SA},)-| \/`, which is pulled straight from strategies.txt:

```
.\geneva-cli.exe intercept -interface "vEthernet (Virtual Switch - WiFi Adapter)" -strategyFile .\from-strategies.txt
I: 14:34:17 outbound \/ inbound {[] ["[TCP:flags:S]-duplicate(tamper{TCP:flags:replace:SA}(send,),)-|"]}
I: 14:34:17 starting up
I: 14:34:17 starting statistics thread
I: 14:34:17 intercepting traffic on vEthernet (Virtual Switch - WiFi Adapter) (idx 11)
I: 14:34:17 using filter "ip and !loopback and ifIdx == 11"
I: 14:34:17 opening handle to WinDivert
E: 14:34:22 failed to send packet after error applying strategy: %!w(<nil>)
I: 14:34:22 intercepted/injected packets: 67/67 inbound, 87/88 outbound; 1 errors
E: 14:34:22 failed to send packet after error applying strategy: %!w(<nil>)
... same line 14 more times ...
I: 14:34:26 stopping all threads
I: 14:34:26 closing WinDivert handle
I: 14:34:26 intercepted/injected packets: 820/820 inbound, 756/772 outbound; 16 errors
```

I was also unable to verify that the strategy specified in s.txt was actually applied as expected.  How did you verify this?

Reviewer: The program also panics if the strategy file does not exist.

Author: So the code expects one strategy passed to it in a file, strategies.txt is supposed to be a reference of known useful strategies. I got it from the main Geneva repository. I should make clear in the readme that passing strategies.txt to intercept is a bad idea.

For the panic, I will update the code to check for the file first and output an error.

> You can run `go test` to run some unit tests, currently there is only tests for validating strategies.txt

Reviewer: There are no tests...
```
go test
?       github.com/getlantern/geneva-cli        [no test files]
```

Author: I forgot to commit it

## How to install as service

Open PowerShell in administrator mode before running any commands

### Install Command, in manual mode

> ```New-Service -Name "Geneva-CLI" -DisplayName "WinGeneva" -Description "Geneva CLI for Windows" -StartupType Manual -BinaryPathName "Path-To-Geneva.exe --args"```

Reviewer: Bit of a nit, but I don't think the service itself should be called "Geneva-CLI"; CLI refers to the command-line program (which I understand the service is utilizing).  You could just call it "Geneva".  Same goes for the description.

### Start

> ```Start-Service -Name "Geneva-CLI"```

Reviewer: This did not work for me.  I ran all instructions as specified in the README in this branch.  Did I screw something up?
```
> New-Service -Name "Geneva-CLI" -DisplayName "WinGeneva" -Description "Geneva CLI for Windows" -StartupType Manual -BinaryPathName "C:\Users\Harry\Code\Lantern\geneva-cli\geneva-cli.exe intercept -interface 'vEthernet (Virtual Switch - WiFi Adapter)' -strategyFile C:\Users\Harry\Code\Lantern\geneva-cli\s.txt"

Status   Name               DisplayName
------   ----               -----------
Stopped  Geneva-CLI         WinGeneva

> Start-Service -Name "Geneva-CLI"
Start-Service : Service 'WinGeneva (Geneva-CLI)' cannot be started due to the following error:
Cannot start service Geneva-CLI on computer '.'.
At line:1 char:1
+ Start-Service -Name "Geneva-CLI"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OpenError: (System.ServiceProcess.ServiceController:ServiceController)
    [Start-Service], ServiceCommandException
    + FullyQualifiedErrorId : CouldNotStartService,Microsoft.PowerShell.Commands.StartServiceCommand
```

Reviewer 2: The System or Application log may have more info?

Author: Uhh double check that PS is in administrator mode?

`.\geneva-cli.exe intercept --interface <interface-name> -strategyFile .\s.txt`

or

`.\geneva-cli.exe intercept --interface <interface-name> -strategy "[TCP:flags:PA]-fragment{tcp:-1:True}-| \/"`

[TCP:flags:PA]-fragment{tcp:8:True}(,fragment{tcp:4:True})-| \/

You can find a list of available interfaces using

`.\geneva-cli.exe list-adapters`

### How to find strategies
You can find a list of strategies in strategies.txt or in the [Geneva paper](http://geneva.cs.umd.edu/papers/geneva_ccs19.pdf) on page 9 with tested success rates for China, India, and Kazakhstan.

## How to test

> You can test or rather verify that the intercept mode is running by simply running it with a validated strategy. The output should show that packets are being rerouted. You can monitor your adapter in [WireShark](https://www.Wireshark.org/) and you should notice a huge uptick in packets being sent from your adapter.

Reviewer 2: ```suggestion
You can test or rather verify that the intercept mode is running by simply running it with a validated strategy. The output should show that packets are being rerouted. You can monitor your adapter in [Wireshark](https://www.wireshark.org/) and you should notice a huge uptick in packets being sent from your adapter.
```

Author: So funnily this happened because I did a find and replace to capitalize the 'W' elsewhere

You can predict what the Wireshark output should look like using the [Geneva documentation](https://github.com/getlantern/geneva?tab=readme-ov-file#strategies-forests-and-action-trees), each strategy will look different. If you take the following strategy and use Wireshark or the included pcap function you will notice that outbound [PSH, ACK] packets will be split into two packets. The pcap feature will show exactly which packets split and into how many.

```[TCP:flags:PA]-fragment{tcp:5:True}-| \/```

> If you are correlating the include pcap output with Wireshark, the Wireshark numbers packets starting at 1 and not 0

Reviewer 2: ```suggestion
If you are correlating the include pcap output with Wireshark, be aware that Wireshark numbers packets start at 1 and not 0.
```

```.\geneva-cli.exe intercept --service stop```

> ### Uninstall``````

Reviewer 2: ```suggestion
### Uninstall
```
